src/reactor.cc

```diff
@@ -112,12 +112,24 @@ void Reactor::Tick() {
     // record the last time series entry if the reactor was operating at the
     // time of retirement.
     if (exit_time() == context()->time()) {
-      if (cycle_step > 0 && cycle_step <= cycle_time &&
+      if (refuel_time == 0){
+        if (cycle_step > 0 && cycle_step <= cycle_time &&
```

Comment:
I thought that the only change you would need to make would be to change "<=" to "<". Did you try that? I *think* that avoids having to say anything about refuel_time (your whole second layer of if-else).

Reply:
without the filter for refuel_time = 0, this case would be wrong:

reactor has lifetime of 6,
cycle_time of 3, and refuel time of 0,
months 1-5: operational
month 6: should operate and generate power.

Without the filter, it would not produce power at month 6 (Because cycle_step=cycle_time, not less), although it should be operating.

Reply:
Does your test pass for my solution (that is, the only change you make is to replace that "<=" with "<")? 


If it does, then your test isn't enough to defend against the bug. Add another test that fails (e.g. the example you've described above.).